pkg/networking/networking.go

```diff
@@ -23,55 +23,43 @@ const (
 
 // NetworkAccess is the interface for network access.
 type NetworkAccess interface {
+	// AddDefaultHeader adds the default headers request.
+	AddDefaultHeader(request *http.Request) error
 	// GetDefaultHeader returns the default header for a given URL.
 	GetDefaultHeader(url *url.URL) http.Header
-	// GetRoundtripper returns the http.Roundtripper.
-	GetRoundtripper() http.RoundTripper
+	// GetRoundTripper returns the http.RoundTripper.
+	GetRoundTripper() http.RoundTripper
 	// GetHttpClient returns the http client.
 	GetHttpClient() *http.Client
 	// AddHeaderField adds a header field to the default header.
 	AddHeaderField(key string, value string)
 	// AddRootCAs adds the root CAs from the given PEM file.
 	AddRootCAs(pemFileLocation string) error
+	// GetAuthenticator returns the authenticator.
+	GetAuthenticator() auth.Authenticator
 }
 
 // NetworkImpl is the default implementation of the NetworkAccess interface.
 type NetworkImpl struct {
-	config       configuration.Configuration
-	userAgent    string
-	staticHeader http.Header
-	logger       *log.Logger
-	proxy        func(req *http.Request) (*url.URL, error)
-	caPool       *x509.CertPool
+	config        configuration.Configuration
+	userAgent     string
+	staticHeader  http.Header
+	logger        *log.Logger
+	proxy         func(req *http.Request) (*url.URL, error)
+	caPool        *x509.CertPool
+	authenticator auth.Authenticator
 }
 
-// customRoundtripper is a custom http.RoundTripper which decorates the request with default headers.
-type customRoundtripper struct {
-	encapsulatedRoundtripper *http.Transport
+// customRoundTripper is a custom http.RoundTripper which decorates the request with default headers.
+type customRoundTripper struct {
+	encapsulatedRoundTripper http.RoundTripper
 	networkAccess            NetworkAccess
-	proxyAuthenticator       *httpauth.ProxyAuthenticator
-}
-
-// decorateRequest appends request header's to the customRoundtripper's instance.
-func (crt *customRoundtripper) decorateRequest(request *http.Request) *http.Request {
-	defaultHeader := crt.networkAccess.GetDefaultHeader(request.URL)
-
-	// iterate over default headers and add them if there is no existing entry yet
-	for k, v := range defaultHeader {
-		if _, found := request.Header[k]; found == false {
-			for i := range v {
-				request.Header.Add(k, v[i])
-			}
-		}
-	}
-
-	return request
 }
 
 // RoundTrip is an implementation of the http.RoundTripper interface.
-func (crt *customRoundtripper) RoundTrip(request *http.Request) (*http.Response, error) {
-	request = crt.decorateRequest(request)
-	return crt.encapsulatedRoundtripper.RoundTrip(request)
+func (crt *customRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	_ = crt.networkAccess.AddDefaultHeader(request)
```

Comment:
Why do we ignore errors here? Shouldn't an error adding a header be handled?

Reply:
I looked at the code and we never return an error, so I refactored the error return value out of it